lapce-data/src/data.rs

```diff
@@ -1596,6 +1596,41 @@ impl LapceTabData {
                     }
                 }
             }
+            LapceWorkbenchCommand::SourceControlDiscardTargetFileChanges => {
+                if let Ok(v) = serde_json::from_value::<FileDiff>(data.unwrap()) {
+                    match v {
+                        FileDiff::Added(path) => {
+                            self.proxy.proxy_rpc.trash_path(
+                                path,
+                                Box::new(move |res| {
+                                    if let Err(err) = res {
+                                        // TODO: inform the user through a corner-notif
+                                        log::warn!(
+                                            "Failed to trash path: {:?}",
+                                            err
+                                        );
+                                    }
+                                }),
+                            );
+                        }
+                        FileDiff::Deleted(path) => {
+                            self.proxy
+                                .proxy_rpc
+                                .git_discard_files_changes(vec![path]);
+                        }
+                        FileDiff::Modified(path) => {
+                            self.proxy
+                                .proxy_rpc
+                                .git_discard_files_changes(vec![path]);
+                        }
+                        FileDiff::Renamed(old_path, new_path) => {
+                            // NOT SURE HOW TO DO THIS ONE YET
```

Comment:
@dzhou121 

Not sure about this one - should we delete the new path + discard changes on old path?